docs/topics/asyncio.rst

```diff
@@ -39,5 +39,28 @@ You can also use custom asyncio event loops with the asyncio reactor. Set the
 :setting:`ASYNCIO_EVENT_LOOP` setting to the import path of the desired event loop class to
 use it instead of the default asyncio event loop.
 
+.. _asyncio-await-dfd:
 
+Awaiting on Deferreds
+=====================
 
+When the asyncio reactor isn't installed, you can await on Deferreds in the
+coroutines directly. When it is installed, this is not possible anymore, due to
+specifics of the Scrapy coroutine integration (the coroutines are wrapped into
+asyncio Futures, not into Deferreds directly), and you need to wrap them into
+Futures. Scrapy provides two helpers for this:
+
+.. autofunction:: scrapy.utils.defer.deferred_to_future
+.. autofunction:: scrapy.utils.defer.maybe_deferred_to_future
+
+If you want to write universal code that works on any reactors,
+you should use ``maybe_deferred_to_future`` on all Deferreds::
+
+    from scrapy.utils.defer import maybe_deferred_to_future
+
+    class MySpider(Spider):
+        # ...
+        async def parse_with_deferred(self, response):
+            additional_response = await maybe_deferred_to_future(treq.get('https://additional.url'))
+            additional_data = await maybe_deferred_to_future(treq.content(additional_response))
```

Comment:
This looks cool, nice to have replacement for inline requests, but will I have to always make 2 deferreds and 2 await to get response? In my project when I use inline_requests I need response 99% of times is there some way to reduce API to

```python
additional_response = await maybe_deferred_to_future(treq.get('https://additional.url'))
```
instead of one await for request and one await for reading response?

Reply:
I think that's just the treq API, you can make a wrapper for it if you really want.

Reply:
+1 to support awaiting Requests, but I think that's a separate task. We'd need to decide on the API, how it works with middlewares, how are exceptions handled, etc.